cpp/test/backends/protocol/otf_message_test.cc

```diff
@@ -0,0 +1,205 @@
+#include <gtest/gtest.h>
+
+#include "mock_socket.hh"
+
+namespace torchserve {
+  TEST(OTFMessageTest, TestEncodeLoadModelResponse) {
+    std::string message = "model_loaded";
+    auto dummyResponse = std::make_unique<torchserve::LoadModelResponse>(
+        200, message.size(), message);
+    char *data = new char[sizeof(LoadModelResponse)];
+    torchserve::OTFMessage::EncodeLoadModelResponse(std::move(dummyResponse), data);
+    const char* expectedResponse = "\x00\x00\x00\xc8\x00\x00\x00\x0cmodel_loaded\xff\xff\xff\xff";
+    EXPECT_TRUE(0 == std::memcmp(data, expectedResponse, sizeof(data)));
+  }
+
+  TEST(OTFMessageTest, TestUTF8EncodeLoadModelResponse) {
+    std::string message = "测试";
+    auto dummyResponse = std::make_unique<torchserve::LoadModelResponse>(
+        200, message.size(), message);
+    char *data = new char[sizeof(LoadModelResponse)];
+    torchserve::OTFMessage::EncodeLoadModelResponse(std::move(dummyResponse), data);
+    const char* expectedResponse = "\x00\x00\x00\xc8\x00\x00\x00\x06\xe6\xb5\x8b\xe8\xaf\x95\xff\xff\xff\xff";
+    EXPECT_TRUE(0 == std::memcmp(data, expectedResponse, sizeof(data)));
+  }
+
+  TEST(OTFMessageTest, TestRetrieveMsgLoadGpu) {
+    MockSocket *client_socket = new MockSocket();
```

Comment:
pls avoid using new as possible as you can. And also you can think of using google test fixtures to setup socket mocker to simplify the source code.

Reply:
Fixed in #1814